pandas/core/internals.py

```diff
@@ -1562,6 +1581,10 @@ def _nanpercentile(values, q, axis, **kw):
         result = self._try_coerce_result(result)
         if is_scalar(result):
             return ax, self.make_block_scalar(result)
+
```

Comment:
again this should be a sparse array, if its not then you need to implement this on the SparseBlock itself (and you can certainly call the super method). I don't like type checking *inside* a block.